lib/trivial.nix

```diff
@@ -467,6 +471,41 @@ rec {
     then v
     else k: v;
 
+  /*
+    Make a function lazy in its arguments.
+
+    Normally when you call a function like `{ a, b }: { r = foo a b; }`,
+    the attribute names of the argument must be evaluated before the evaluator
+    can return a the (partially evaluated) function body (`{ r = ...; }`).
+
+    `lazyFunction` uses reflection on the passed in function and it only works
+    for functions defined with a "set pattern".
```

Comment:
It using reflection (which is not really a known term) is an implementation detail, can be omitted. And the note about it only working with set patterns should be moved down.

```suggestion
```

Reply:
I think it's relevant because using reflection breaks the function abstraction, which is a big deal.
I've removed it for now, because I agree describing the consequences is more important.